### PR TITLE
api(win_config): add nightly open win split support

### PR DESCRIPTION
**Pull request by myypo**
Noticed that new nightly functionality for `nvim_open_win`  that allows to open regular splits is not possible to use with the crate yet. Hopefully it will be of help. 

Also made `WindowRelativeTo` use serde untagged instead of the custom implementation, since there was a comment linking to [this issue](https://github.com/serde-rs/serde/issues/1402).

Please, let me know if I missed something.

### CODE REVIEW

File: tests/src/api/win_config.rs [hunk 1]
```diff
@@ -77,6 +77,35 @@ fn open_win_full_config() {
...
+    let win = res.unwrap();
+
+    let got = win.get_config();
+    assert!(got.is_ok(), "{got:?}");
+
+    let got = got.unwrap();
+    // `vim.api.nvim_win_get_config` does not seem
+    // to ever return `vertical` field
+    // assert_eq!(config.vertical, got.vertical);
+    assert_eq!(config.split, got.split);
```
Comment (noib3) [inline tests/src/api/win_config.rs:103]: I believe we can unwrap both of these to make sure the direction is `Some(SplitDirection::Right)`.

File: crates/api/src/types/window_relative_to.rs [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
-    Window(Window),
-
     /// Positions the window relative to the current cursor position.
     Cursor,
 
     /// Positions the window relative to the current mouse cursor position..
     Mouse,
+
+    /// Positions the window relative to another window.
+    #[serde(untagged)]
```
Comment (noib3) [inline crates/api/src/types/window_relative_to.rs:20]: It seems like this is making the `open_win_basic_config` test fail on 0.8 and 0.9.

**Review by noib3 — CHANGES_REQUESTED** [review]

**Comment by myypo** [comment]
I've resolved the non-nightly variant of `open_win_basic_config` test failing. Nightly was ok since it does not rely on [this implementation](https://github.com/noib3/nvim-oxi/blob/8d713660115fee32a0ae724e10a79fb2d6978f30/crates/api/src/types/window_relative_to.rs#L37)

Also added unwraps to all options in `win_config.rs` integration test.

Nightly CI seems to get stuck on integration tests both on this PR and on HEAD as I checked on my fork https://github.com/myypo/nvim-oxi. Will, probably, look into it later.

**Comment by myypo** [comment]
The PR that broke the CI: https://github.com/neovim/neovim/pull/28432 

The 3 functions in question were marked as experimental so they are likely to change in the future again.

**Comment by noib3** [comment]
Great work, thanks.

**Pull request merged**